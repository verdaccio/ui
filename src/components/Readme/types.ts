export interface IProps {
  description: string;
}
