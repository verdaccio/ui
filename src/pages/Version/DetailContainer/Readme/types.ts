export interface Props {
  description: string;
}
