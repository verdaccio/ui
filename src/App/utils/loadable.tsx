import React, { lazy, Suspense } from 'react';
export default (importCallback: any) => {
  const TargetComponent = lazy(importCallback);
  return (props: any) => (
    <Suspense fallback={null}>
      <TargetComponent {...props} />
    </Suspense>
  );
};
