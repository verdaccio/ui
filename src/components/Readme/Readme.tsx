import React from 'react';
import 'github-markdown-css';

import { Props } from './types';

const Readme: React.FC<Props> = ({ description }) => <div className="markdown-body" dangerouslySetInnerHTML={{ __html: description }} />;

export default Readme;
