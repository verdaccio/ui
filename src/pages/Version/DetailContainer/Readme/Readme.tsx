import styled from '@emotion/styled';
import React from 'react';
import 'github-markdown-css';

import { Theme } from 'verdaccio-ui/design-tokens/theme';

import { Props } from './types';

const Readme: React.FC<Props> = ({ description }) => (
  <Wrapper className="markdown-body" dangerouslySetInnerHTML={{ __html: description }} />
);

export default Readme;

const Wrapper = styled('div')<{ theme?: Theme }>(({ theme }) => ({
  background: theme?.palette.white,
  color: theme?.palette.black,
  padding: theme?.spacing(2, 3),
  ul: {
    listStyle: 'disc',
  },
}));
