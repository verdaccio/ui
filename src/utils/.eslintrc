{
    "rules": {
        "no-invalid-this": 0
    }
}