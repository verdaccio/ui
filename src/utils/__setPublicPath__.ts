// @ts-ignore
if (!__DEBUG__) {
  // @ts-ignore
  __webpack_public_path__ = window.VERDACCIO_API_URL.replace(/\/verdaccio\/$/, '/static/'); // eslint-disable-line
}
