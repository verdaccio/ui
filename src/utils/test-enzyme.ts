import { mount, shallow } from 'enzyme';
import React from 'react';

import ThemeProvider from 'verdaccio-ui/design-tokens/ThemeProvider';

const shallowWithTheme = (element: React.ReactElement<any>, ...props: any[]) =>
  shallow(element, {
    wrappingComponent: ThemeProvider,
    ...props,
  });

const mountWithTheme = (element: React.ReactElement<any>, ...props: any[]): any =>
  mount(element, {
    wrappingComponent: ThemeProvider,
    ...props,
  });

export { mountWithTheme as mount, shallowWithTheme as shallow };
