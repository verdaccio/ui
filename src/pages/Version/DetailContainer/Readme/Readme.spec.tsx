import React from 'react';

import { mount } from 'verdaccio-ui/utils/test-enzyme';

import Readme from './Readme';

describe('<Readme /> component', () => {
  test('should load the component in default state', () => {
    const wrapper = mount(<Readme description="test" />);
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('should dangerously set html', () => {
    const wrapper = mount(<Readme description="<h1>This is a test string</h1>" />);
    expect(wrapper.html()).toEqual(
      '<div class="markdown-body css-beaqbv-Wrapper esdleg90"><h1>This is a test string</h1></div>'
    );
    expect(wrapper.html()).toMatchSnapshot();
  });
});
