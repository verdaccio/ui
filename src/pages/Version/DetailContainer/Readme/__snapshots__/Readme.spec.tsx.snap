// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Readme /> component should dangerously set html 1`] = `"<div class=\\"markdown-body css-beaqbv-Wrapper esdleg90\\"><h1>This is a test string</h1></div>"`;

exports[`<Readme /> component should load the component in default state 1`] = `"<div class=\\"markdown-body css-beaqbv-Wrapper esdleg90\\">test</div>"`;
